src/mongocxx/test/collection.cpp

```diff
@@ -2808,4 +2812,103 @@ TEST_CASE("Ensure that the WriteConcernError 'errInfo' object is propagated", "[
     REQUIRE(contains_err_info);
 }
 
+TEST_CASE("expose writeErrors[].errInfo", "[collection]") {
+    bool insert_succeeded = false;  // this is checked by side-effect
+
+    // A helper for checking that an error document is well-formed according to our requirements:
+    auto writeErrors_well_formed =
+        [&insert_succeeded](const bsoncxx::document::view& reply_view) -> bool {
```

Comment:
The `insert_succeeded` capture is no longer used.

This is causing a test failure on [this task](https://spruce.mongodb.com/task/cxx_driver_ubuntu1804_debug_asan_50_compile_and_test_with_static_libs_extra_alignment_patch_a7be2f22f2cd4c8664e48abae4f78b6f1c9d6e99_621cfd0fc9ec4457e1f6d574_22_02_28_16_49_21/logs?execution=0):

```
[2022/02/28 18:24:14.868] found libmongoc version 1.22.0-pre
[2022/02/28 18:24:14.868] /data/mci/641d255fa2451925acd2c0de5d0c083e/mongo-cxx-driver/src/mongocxx/test/collection.cpp:2820:11: error: lambda capture 'insert_succeeded' is not used [-Werror,-Wunused-lambda-capture]
[2022/02/28 18:24:14.868]         [&insert_succeeded](const bsoncxx::document::view& reply_view) -> bool {
```

Reply:
Sorry to have missed this, I thought I had looked at all of them-- I had not! I'm also going to try to see why I'm not seeing this locally... thank you for your patience.

Reply:
Looks like my local CXX build flags are extremely permissive compared to my CFLAGS-- modifying.